GrandCentralBoard/Widgets/Bonus/RequestBuilder.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  RequestBuilder.swift
+//  GrandCentralBoard
+//
+//  Created by Rafal Augustyniak on 06/04/16.
+//
+
+import Foundation
+
+
+enum RequestBuilderError : ErrorType {
+    case IncorrectBaseURL(URL: NSURL)
+    case IncorrectResultURL(baseURL: NSURL, path: String)
+}
+
+
+final class RequestBuilder<T: RequestTemplateProtocol> {
+
+    private let requestTemplate: T
+    
+    init(requestTemplate: T) {
+        self.requestTemplate = requestTemplate
+    }
+
+    func buildURLRequest() throws -> NSURLRequest {
+        let URL = try createURL()
+
+        let URLRequest = NSMutableURLRequest.init(URL: URL)
+        URLRequest.HTTPBody = try createHTTPBody()
+        URLRequest.HTTPMethod = self.requestTemplate.method.HTTPMethod
+        URLRequest.setValue("application/json;charset=UTF-8", forHTTPHeaderField: "Content-Type")
+        return URLRequest
+    }
+
+    private func createHTTPBody() throws -> NSData? {
+        switch requestTemplate.method {
+        case .Get(_):
+            return nil
+        case .Post(let bodyParameters, _):
+            return try NSJSONSerialization.dataWithJSONObject(bodyParameters, options: NSJSONWritingOptions.init(rawValue: 0))
+        }
+    }
+
+    private func createURL() throws -> NSURL {
+        guard let URLComponents = NSURLComponents.init(URL: self.requestTemplate.baseURL, resolvingAgainstBaseURL: true) else {
+            throw RequestBuilderError.IncorrectBaseURL(URL: self.requestTemplate.baseURL)
```

Comment:
I don't like this class - basically all of this could be handled by Alamofire (it is tested and we know it is good for networking). But we could refactor this later, after changes from #51